File: IAIso-v5.0/vision/sdk/csharp/IAIso.Core/Engine.cs
using System;

namespace IAIso.Core {
    public class Engine {
        private double _p = 0.0;
        public bool BackProp { get; set; } = true;

        public string Update(int tokens, int tools) {
            double delta = (tokens * 0.00015) + (tools * 0.08);
            _p = Math.Max(0, _p + delta - 0.02);
            if (_p >= 0.95) { _p = 0.0; return "RELEASED"; }
            return _p >= 0.85 ? "ESCALATED" : "OK";
        }
    }
}

File: IAIso-v5.0/vision/sdk/go/iaiso/engine.go
package iaiso

import "math"

type Engine struct {
    SystemID string
    Pressure float64
    BackProp bool
}

func NewEngine(id string) *Engine {
    return &Engine{SystemID: id, Pressure: 0.0, BackProp: true}
}

func (e *Engine) Update(tokens int, tools int) string {
    delta := (float64(tokens) * 0.00015) + (float64(tools) * 0.08)
    e.Pressure = math.Max(0, e.Pressure + delta - 0.02)

    if e.Pressure >= 0.95 {
        e.Pressure = 0.0
        return "RELEASE_TRIGGERED"
    }
    return "OK"
}
